### PR TITLE
feat: ensure log order

### PR DESCRIPTION
**Pull request by EstaniPP**
This PR manages to solve the log order issue. It uses a priority queue in the plugin which will dispatch the log only if the previous ones were already dispatched.

### CODE REVIEW

**Comment by patak-dev** [comment]
Hey @EstaniPP, thanks for the PR. The current implementation isn't going to work correctly if the user has several pages opened against the same dev server. A few needed changes:
1. remove the new `restartQueue` command
2. use a URL for the commands as `__terminal/log?m=...&t=...`
3. the t can be `Date.now()`, so concurrent requests are allowed
4. you can parse the search using [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) in node
5. we don't need a complex queque. Every time a new request happens, place it in an array and set a timeout of 50ms (we can check the best time), and after it check the array for the min t and log it if it is less this log t, repeat with the next min until you reach the current log and stop (the next timeout will kick in and log the following ones)

**Comment by EstaniPP** [comment]
@patak-dev dev, requested changes done! Please review it

File: src/logQueue.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      return first.priority - second.priority
+    else
+      return first.queueOrder - second.queueOrder
+  })
+  if (priority < lastDispatched) {
+    setTimeout(() => {
+      while (queue[0] && queue[0].priority < lastDispatched) {
+        const dispatchFunction = queue.shift()?.dispatchFunction
+        if (dispatchFunction)
+          dispatchFunction()
```
Comment (patak-dev) [inline src/logQueue.ts:23]: We can use `!` here
```suggestion
        queue.shift()!.dispatchFunction()
```

**Review by patak-dev — APPROVED** [review]
Awesome! Thanks again :)

**Pull request merged**